src/main/java/net/schmizz/sshj/SSHClient.java

```diff
@@ -139,6 +139,8 @@
     /** {@code ssh-connection} service */
     protected final Connection conn;
 
+    private List<LocalPortForwarder> forwarders;
```

Comment:
Make final and initialize with `new ArrayList<>()`? This prevents the excessive null checks.